### PR TITLE
Fix for group samples test can sort the list of samples

### PR DESCRIPTION
**Pull request by ChrisHuynh333**
## What does this PR do and why?
The group samples test 'can sort the list of samples' would sometimes error. This is because the PUIDs generated for testing are not always the same, and therefore, the samples ordering when sorting by PUID would sometimes change.

Rather than expecting specific samples in a certain order, we now gather the first four PUIDs of the table listing, and compare them against one another to ensure they are listed properly.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
1. Ensure tests pass locally

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: test/system/groups/samples_test.rb [hunk 1]
```diff
@@ -16,6 +16,16 @@ def setup
       @sample31 = samples(:sample31)
     end
 
+    def retrieve_puids
+      within first('table tbody#group-samples-table-body') do
+        puids = []
+        puids << first('tr:nth-child(1) td:first-child').text
```
Comment (joshsadam) [inline test/system/groups/samples_test.rb:22]: **suggestion:** Simplify this by putting it in a loop

Comment (ChrisHuynh333) [reply test/system/groups/samples_test.rb]: Changed in a5a08729ba05d4d32eaf14bd0833e445db4b336f

File: test/system/groups/samples_test.rb [hunk 2]
```diff
@@ -95,22 +104,22 @@ def setup
...
 
+      # Because PUIDs are not always generated the same, issues regarding order have occurred when hard testing
+      # the expected ordering of samples based on PUID. To resolve this, we will gather the first 4 PUIDs and ensure
+      # they are ordered as expected against one another.
       assert_selector 'table tbody#group-samples-table-body tr', count: 20
-      within first('table tbody#group-samples-table-body') do
-        assert_selector 'tr:first-child td:first-child', text: @sample1.puid
-        assert_selector 'tr:first-child td:nth-child(2)', text: @sample1.name
+      puids = retrieve_puids
+      3.times do |n|
```
Comment (joshsadam) [inline test/system/groups/samples_test.rb:114]: **nitpick:** Instead of hard coding the 3, let do `(puids.length -1)`

Comment (ChrisHuynh333) [reply test/system/groups/samples_test.rb]: Changed in 0cd4235004b2869291f8b90952bf976b2c908388

Comment (joshsadam) [reply test/system/groups/samples_test.rb]: Awesome, thanks

**Review by joshsadam — APPROVED** [review]
Looks great to me, and passes locally.

**Pull request merged**